RapidWebView/src/main/java/in/workindia/rapidwebview/utils/RapidDownLoadUtility.kt

```diff
@@ -0,0 +1,165 @@
+package `in`.workindia.rapidwebview.utils
+
+import android.app.DownloadManager
+import android.content.ActivityNotFoundException
+import android.content.ContentResolver
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.Environment
+import android.widget.Toast
+import androidx.core.content.FileProvider
+import `in`.workindia.rapidwebview.datamodels.DownloadLocation
+import `in`.workindia.rapidwebview.R
+import `in`.workindia.rapidwebview.constants.BroadcastConstants.Companion.PROVIDER_SUFFIX
+import `in`.workindia.rapidwebview.utils.Utils.validateUrl
+import java.io.File
+
+object RapidDownLoadUtility {
+
+    /**
+     * Starts a download using the system's `DownloadManager`.
+     *
+     * @param url The URL from which the file is to be downloaded. Must be a valid URL.
+     * @param context The context, used to access system services.
+     * @param fileName The name of the file to be downloaded. If null, the name will be extracted from the URL. Defaults to null.
+     * @param downloadLocation The location where the file will be saved. Defaults to [DownloadLocation.PUBLIC_DOWNLOADS].
+     * @return `true` if the download was successfully initiated, `false` otherwise.
+     */
+    fun startDownload(
+        url: String?,
```

Comment:
`context` should be the first parameter.

Why would the url and filename be null safe?